lib/chef/provider/service/systemd.rb

```diff
@@ -76,6 +76,31 @@ def define_resource_requirements
     end
   end
 
+  def systemd_service_status
+    # If the status has already been collected, return that
+    return @service_status unless @service_status.nil?
+
+    # Collect all the status information for a service and returns it at once
+    options, args = get_systemctl_options_args
+    s = shell_out!("#{systemctl_path} #{args} show -p UnitFileState -p ActiveState #{new_resource.service_name}", options)
+    # e.g. /bin/systemctl --system show  -p UnitFileState -p ActiveState sshd.service
+    # Returns something like:
+    # ActiveState=active
+    # UnitFileState=enabled
+    status = {}
+    s.stdout.each_line do |line|
+      k, v = line.strip.split("=")
+      status[k] = v
+    end
+    # Assert requisite keys exist
+    unless status.key?("UnitFileState") && status.key?("ActiveState")
+      raise Mixlib::ShellOut::ShellCommandFailed, "'#{systemctl_path} show' not reporting status for #{new_resource.service_name}!"
```

Comment:
I don't think raising a Mixlib::Shellout exception here is the right thing to do. Using that exception class implies that the error came from Mixlib::Shellout which isn't the case. With no better option, a RuntimeException (ie, a bare call to `raise` is probably what you want here.

Reply:
My intent here is to surface that the shellout to systemctl failed; if `RuntimeException` is the preference, that works for me.

Reply:
@tas50 @lamont-granquist thoughts on this?

Reply:
I'm going to change this to `raise Chef::Exceptions::Service` per Lamont's suggestion